sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_logical_subscriptions_history_v1/query.sql

```diff
@@ -0,0 +1,121 @@
+WITH plan_services AS (
+  SELECT
+    plan_id,
+    ARRAY_AGG(
+      STRUCT(services.id, services.name, tier.name AS tier)
+      ORDER BY
+        services.id
+    ) AS services
+  FROM
+    `moz-fx-data-shared-prod.subscription_platform_derived.services_v1` AS services
+  CROSS JOIN
+    UNNEST(services.stripe_plan_ids) AS plan_id
+  LEFT JOIN
+    UNNEST(services.tiers) AS tier
+  ON
+    plan_id IN UNNEST(tier.stripe_plan_ids)
+  GROUP BY
+    plan_id
+),
+paypal_subscriptions AS (
+  SELECT DISTINCT
+    subscription_id
+  FROM
+    `moz-fx-data-shared-prod.stripe_external.invoice_v1`
+  WHERE
+    JSON_VALUE(metadata, '$.paypalTransactionId') IS NOT NULL
+)
+SELECT
+  CONCAT(
+    'Stripe-',
+    history.subscription.id,
+    '-',
+    subscription_item.id,
+    '-',
+    FORMAT_TIMESTAMP('%FT%H:%M:%E6S', history.subscription.start_date),
+    '-',
+    FORMAT_TIMESTAMP('%FT%H:%M:%E6S', history.valid_from)
+  ) AS id,
+  history.valid_from,
+  history.valid_to,
+  history.id AS provider_subscriptions_history_id,
+  STRUCT(
+    CONCAT(
+      'Stripe-',
+      history.subscription.id,
+      '-',
+      subscription_item.id,
+      '-',
+      FORMAT_TIMESTAMP('%FT%H:%M:%E6S', history.subscription.start_date)
+    ) AS id,
+    'Stripe' AS provider,
+    IF(paypal_subscriptions.subscription_id IS NOT NULL, 'PayPal', 'Stripe') AS payment_provider,
+    history.subscription.id AS provider_subscription_id,
+    subscription_item.id AS provider_subscription_item_id,
+    history.subscription.created AS provider_subscription_created_at,
+    history.subscription.customer_id AS provider_customer_id,
+    history.customer.metadata.userid AS mozilla_account_id,
+    history.customer.metadata.userid_sha256 AS mozilla_account_id_sha256,
+    COALESCE(
+      NULLIF(history.customer.shipping.address.country, ''),
+      NULLIF(history.customer.address.country, '')
+      -- TODO: also use charge billing addresses
+    ) AS country_code,
```

Comment:
This probably should use the same [bifurcated logic that `stripe_subscriptions_history_v1` does](https://github.com/mozilla/bigquery-etl/blob/aa49fd4c383b2348ec1726672549c6e1d71b4b7f/sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_subscriptions_history_v1/query.sql#L356-L373) to calculate the country differently before vs after we enabled Stripe Tax on 2022-12-01, but I figured that should wait until I incorporated the charge billing addresses (otherwise the country would always end up null for records before 2022-12-01).

Reply:
Never mind.  It turned out not using charge/card addresses resulted in 14% of currently active subscriptions missing a country, which was going to be distracting, so I went ahead and implemented that plus the aforementioned Stripe Tax-related address logic.